### PR TITLE
feat: add playwright tests to check for zero summary of past emails in sidebar

### PR DESCRIPTION
**Pull request by retrogtx**
# READ CAREFULLY THEN REMOVE

Remove bullet points that are not relevant.

PLEASE REFRAIN FROM USING AI TO WRITE YOUR CODE AND PR DESCRIPTION. IF YOU DO USE AI TO WRITE YOUR CODE PLEASE PROVIDE A DESCRIPTION AND REVIEW IT CAREFULLY. MAKE SURE YOU UNDERSTAND THE CODE YOU ARE SUBMITTING USING AI.

- Pull requests that do not follow these guidelines will be closed without review or comment.
- If you use AI to write your PR description your pr will be close without review or comment.
- If you are unsure about anything, feel free to ask for clarification.

## Description

Please provide a clear description of your changes.

---

## Type of Change

Please delete options that are not relevant.

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature with breaking changes)
- [ ] 📝 Documentation update
- [ ] 🎨 UI/UX improvement
- [ ] 🔒 Security enhancement
- [ ] ⚡ Performance improvement

## Areas Affected

Please check all that apply:

- [ ] Email Integration (Gmail, IMAP, etc.)
- [ ] User Interface/Experience
- [ ] Authentication/Authorization
- [ ] Data Storage/Management
- [ ] API Endpoints
- [ ] Documentation
- [ ] Testing Infrastructure
- [ ] Development Workflow
- [ ] Deployment/Infrastructure

## Testing Done

Describe the tests you've done:

- [ ] Unit tests added/updated
- [ ] Integration tests added/updated
- [ ] Manual testing performed
- [ ] Cross-browser testing (if UI changes)
- [ ] Mobile responsiveness verified (if UI changes)

## Security Considerations

For changes involving data or authentication:

- [ ] No sensitive data is exposed
- [ ] Authentication checks are in place
- [ ] Input validation is implemented
- [ ] Rate limiting is considered (if applicable)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/Mail-0/Zero/blob/staging/.github/CONTRIBUTING.md) document
- [ ] My code follows the project's style guidelines
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in complex areas
- [ ] I have updated the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix/feature works
- [ ] All tests pass locally
- [ ] Any dependent changes are merged and published

## Additional Notes

Add any other context about the pull request here.

## Screenshots/Recordings

Add screenshots or recordings here if applicable.

---

_By submitting this pull request, I confirm that my contribution is made under the terms of the project's license._

    

---

## Summary by cubic
Added Playwright end-to-end tests to verify that the sidebar correctly summarizes the past five emails.

- **Testing**
  - Checks that the AI chat sidebar opens and displays a summary when prompted.
  - Confirms the summary is visible and contains content.

## Summary by CodeRabbit

* **New Features**
  * Added an end-to-end test to verify that the AI chat can summarize recent emails and display the result in the inbox sidebar.

* **Style**
  * Added a data attribute to each AI chat message for improved message role identification.

### CODE REVIEW

File: packages/testing/e2e/ai-summary.spec.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      const welcomeModal = page.getByText('Welcome to Zero Email!');
+      if (await welcomeModal.isVisible({ timeout: 2000 })) {
+        console.log('Onboarding modal detected, dismissing...');
+        await page.locator('body').click({ position: { x: 100, y: 100 } });
+        await page.waitForTimeout(1500);
+        console.log('Modal successfully dismissed');
+      }
+    } catch {
+      console.log('No onboarding modal found, proceeding...');
+    }
```
[the remark this answers is not in the document]
Comment (retrogtx) [reply packages/testing/e2e/ai-summary.spec.ts:26]: we need a hard limit; i tried this before and could take alot time randomly, better to re run incase time is indeed short

File: packages/testing/e2e/ai-summary.spec.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    
+    const assistantMessage = page.locator('[data-message-role="assistant"]').last();
+    await expect(assistantMessage).toBeVisible({ timeout: 15000 });
+    
+    const responseText = await assistantMessage.textContent();
+
+    console.log('AI Response Text:', responseText);
+    expect(responseText).toBeTruthy();
+    expect(responseText!.length).toBeGreaterThan(15);
+
```
[the remark this answers is not in the document]
Comment (retrogtx) [reply packages/testing/e2e/ai-summary.spec.ts:52]: the response can be different each time no need to check for "keywords"

**Comment by MrgSub** [comment]
### Merge activity

* **Aug 7, 4:01 AM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/Mail-0/Zero/1920).
* **Aug 7, 4:02 AM UTC**: @MrgSub merged this pull request with [Graphite](https://app.graphite.dev/github/pr/Mail-0/Zero/1920).
